packages/compile-solidity/src/compilerSupplier/Cache.ts

```diff
@@ -24,13 +26,31 @@ export class Cache {
   add(code: string, fileName: string) {
     const filePath = this.resolve(fileName);
     fs.writeFileSync(filePath, code);
+    this.memoizedCompilers[fileName] = code;
```

Comment:
```suggestion
    this.memoizedCompilers[filePath] = code;
```
To prevent `fileNames` that would resolve to the same path, I think the map key should be normalized, that is the resolved path in this case.